### PR TITLE
feat: CP support Upload className and style

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support Upload className and style |
| 🇨🇳 Chinese |          - |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at eb72f2f</samp>

This pull request adds a new feature to the `ConfigProvider` component, which allows users to customize the style and behavior of the `Upload` component. It also refactors the `Upload` component to use the `ConfigContext` and adds test cases and documentation for the new feature.

### <samp>🤖 Generated by Copilot at eb72f2f</samp>

*  Add a new optional upload prop to the ConfigProvider component, which allows setting the className and style of the Upload component ([link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR124), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R168), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R286), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R366), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R136), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R138))
*  Modify the Upload component to use the upload prop from the ConfigContext to customize the wrapper element, and simplify the render logic for the upload button ([link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-a2e7152a9f97730a600f9bda0bf15029cce0148818a6ee8d2a037fafc101427dL332-R332), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-a2e7152a9f97730a600f9bda0bf15029cce0148818a6ee8d2a037fafc101427dL408-R431), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-a2e7152a9f97730a600f9bda0bf15029cce0148818a6ee8d2a037fafc101427dL446-R460), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-a2e7152a9f97730a600f9bda0bf15029cce0148818a6ee8d2a037fafc101427dL474-R467))
*  Add a new test case to check the upload className and style props in the ConfigProvider component, and import the Upload component for the test ([link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R35), [link](https://github.com/ant-design/ant-design/pull/43325/files?diff=unified&w=0#diff-58612c0e42c2460919a2c73d9dac7eed9d88cb7d97da183b5f94e83ce74b8149R669-R679))

### CODE REVIEW

File: components/upload/Upload.tsx [hunk 1]
```diff
@@ -453,25 +453,14 @@ const InternalUpload: React.ForwardRefRenderFunction<UploadRef, UploadProps> = (
...
-      >
+      <span className={wrapperCls} style={mergedStyle}>
         {renderUploadList(uploadButton, !!children)}
       </span>,
     );
   }
 
   return wrapSSR(
-    <span className={classNames(`${prefixCls}-wrapper`, rtlCls, className, hashId)}>
+    <span className={wrapperCls} style={mergedStyle}>
```
Comment (MadCcc) [inline components/upload/Upload.tsx:463]: 先不要新增，这个 PR 只做 ctx 支持

File: components/upload/Upload.tsx [hunk 2]
```diff
@@ -405,30 +405,30 @@ const InternalUpload: React.ForwardRefRenderFunction<UploadRef, UploadProps> = (
...
+    const dragCls = classNames(hashId, prefixCls, `${prefixCls}-drag`, {
+      [`${prefixCls}-drag-uploading`]: mergedFileList.some((file) => file.status === 'uploading'),
+      [`${prefixCls}-drag-hover`]: dragState === 'dragover',
+      [`${prefixCls}-disabled`]: mergedDisabled,
+      [`${prefixCls}-rtl`]: direction === 'rtl',
+    });
+
     return wrapSSR(
-      <span className={classNames(`${prefixCls}-wrapper`, rtlCls, className, hashId)}>
+      <span className={wrapperCls}>
```
Comment (MadCcc) [inline components/upload/Upload.tsx:425]: 这里用 wrapperCls 多了两条 class

File: components/upload/Upload.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/upload/Upload.tsx:463]: 我删一下

Comment (li-jia-nan) [reply components/upload/Upload.tsx]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**